File: t/PACSFramework/CS/DicomHighlevelPatientUpdaterDemo/Dicom/PatientUpdaterConnection.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Leadtools.Dicom.Scu.Common;
using Leadtools.Dicom;

namespace DicomDemo.Dicom
{
    public class PatientUpdaterConnection : DicomConnection
    {
        private new bool _Rejected = false;

        public new bool Rejected
        {
            get { return _Rejected; }           
        }       

        private string _Reason;

        public string Reason
        {
            get { return _Reason; }            
        }

        protected override void OnReceiveAssociateReject(DicomAssociateRejectResultType result, DicomAssociateRejectSourceType source, DicomAssociateRejectReasonType reason)
        {
            base.OnReceiveAssociateReject(result, source, reason);                       
            _Rejected = true;
            switch (reason)
            {
                case DicomAssociateRejectReasonType.Called:
                    _Reason = "Called AE Title not recognized.";
                    break;
                case DicomAssociateRejectReasonType.Calling:
                    _Reason = "Calling AE Title not recognized.";
                    break;               
                case DicomAssociateRejectReasonType.Congestion:
                    _Reason = "Temporary congestion";
                    break;
                default:
                    _Reason = "Uknown association rejection.";
                    break;
            }
        }
    }
}

File: t/PACSFramework/VB/DicomHighlevelPatientUpdaterDemo/Dicom/PatientUpdaterConnection.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Leadtools.Dicom
Imports Leadtools.Dicom.Scu.Common

Namespace DicomDemo.Dicom
   Public Class PatientUpdaterConnection : Inherits DicomConnection
        Private Shadows _Rejected As Boolean = False

        Public Overloads ReadOnly Property Rejected() As Boolean
            Get
                Return _Rejected
            End Get
        End Property

        Private _Reason As String

      Public ReadOnly Property Reason() As String
         Get
            Return _Reason
         End Get
      End Property

      Protected Overrides Sub OnReceiveAssociateReject(ByVal result As DicomAssociateRejectResultType, ByVal source As DicomAssociateRejectSourceType, ByVal reason As DicomAssociateRejectReasonType)
         MyBase.OnReceiveAssociateReject(result, source, reason)
         _Rejected = True
         Select Case reason
            Case DicomAssociateRejectReasonType.Called
               _Reason = "Called AE Title not recognized."
            Case DicomAssociateRejectReasonType.Calling
               _Reason = "Calling AE Title not recognized."
            Case DicomAssociateRejectReasonType.Congestion
               _Reason = "Temporary congestion"
            Case Else
               _Reason = "Uknown association rejection."
         End Select
      End Sub
   End Class
End Namespace
